src/haz3lcore/zipper/Editor.re

```diff
@@ -54,35 +98,70 @@ module Meta = {
   let yojson_of_t = _ => failwith("Editor.Meta.yojson_of_t");
   let t_of_yojson = _ => failwith("Editor.Meta.t_of_yojson");
 
+  let next_projected = (z_projected: Projector.proj_ret, ~touched, ~old) => {
+    let segment = Zipper.unselect_and_zip(z_projected.z);
+    let (term, terms) = MakeTerm.go(segment);
+    let measured = Measured.of_segment(~touched, ~old, segment);
+    {
+      z: z_projected.z,
+      segment,
+      term,
+      terms,
+      measured,
+      term_ranges: TermRanges.mk(segment),
+      tiles: TileMap.mk(segment),
+      holes: Segment.holes(segment),
+      syntax_map: z_projected.syntax_map,
+    };
+  };
+
+  let next_statics =
+      (
+        ~settings: CoreSettings.t,
+        a: Action.t,
+        z: Zipper.t,
+        old_statics: CachedStatics.t,
+      ) =>
+    if (!settings.statics) {
+      CachedStatics.empty;
+    } else if (!Action.is_edit(a)) {
```

Comment:
I the idea here that we want to cache around cursor movement since the statics shouldn't change but the zipper will have?

Reply:
yes. this optimization hasn't changed, it's just relocated from Update